packages/odf/components/topology/Topology.tsx

```diff
@@ -0,0 +1,512 @@
+
+const Topology: React.FC = () => {
+  const [controller, setController] = React.useState<Visualization>(null);
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [visualizationLevel, setVisualizationLevel] =
+    React.useState<TopologyViewLevel>(TopologyViewLevel.NODES);
+  const [activeItemsUID, setActiveItemsUID] = React.useState<string[]>([]);
+  const [activeItem, setActiveItem] = React.useState<string>('');
+  const [activeNode, setActiveNode] = React.useState('');
+
+  const [nodes, nodesLoaded, nodesError] =
+    useK8sWatchResource<NodeKind[]>(nodeResource);
+
+  const [storageCluster, storageClusterLoaded, storageClusterError] =
+    useK8sWatchResource<StorageClusterKind[]>(storageClusterResource);
+
+  const [deployments, deploymentsLoaded, deploymentsError] =
+    useK8sWatchResource<DeploymentKind[]>(odfDeploymentsResource);
+
+  const [pods, podsLoaded, podsError] =
+    useK8sWatchResource<PodKind[]>(odfPodsResource);
+
+  const [statefulSets, statefulSetLoaded, statefulSetError] =
+    useK8sWatchResource<K8sResourceCommon[]>(odfStatefulSetResource);
+
+  const [replicaSets, replicaSetsLoaded, replicaSetsError] =
+    useK8sWatchResource<K8sResourceCommon[]>(odfReplicaSetResource);
+
+  const [daemonSets, daemonSetsLoaded, daemonSetError] =
+    useK8sWatchResource<K8sResourceCommon[]>(odfDaemonSetResource);
+
+  const odfNodes = nodes.filter((node) =>
+    _.has(node.metadata.labels, CEPH_STORAGE_LABEL)
+  );
+
+  const memoizedNodes = useDeepCompareMemoize(odfNodes, true);
+  const memoizedDeployments = useDeepCompareMemoize(deployments, true);
+  const memoizedPods = useDeepCompareMemoize(pods, true);
+  const memoizedStatefulSets = useDeepCompareMemoize(statefulSets, true);
+  const memoizedReplicaSets = useDeepCompareMemoize(replicaSets, true);
+  const memoizedDaemonSets = useDeepCompareMemoize(daemonSets, true);
+
+  const nodeDeploymentMap = React.useMemo(
+    () =>
+      generateNodeDeploymentsMap(
+        memoizedNodes,
+        memoizedPods,
+        ...memoizedReplicaSets,
+        ...memoizedReplicaSets,
+        ...memoizedDaemonSets,
+        ...memoizedStatefulSets,
+        ...memoizedDeployments
+      ),
+    [
+      memoizedDaemonSets,
+      memoizedDeployments,
+      memoizedNodes,
+      memoizedPods,
+      memoizedReplicaSets,
+      memoizedStatefulSets,
+    ]
+  );
+
+  const onStepInto = (args) => {
+    const nodeName = args.label;
+    setActiveNode(nodeName);
+    setVisualizationLevel(TopologyViewLevel.DEPLOYMENTS);
+  };
+
+  const onStepOut = () => {
+    setActiveNode('');
+    setVisualizationLevel(TopologyViewLevel.NODES);
+  };
+
+  React.useEffect(() => {
+    const temp = new Visualization();
+    temp.registerLayoutFactory(defaultLayoutFactory);
+    temp.registerComponentFactory(stylesComponentFactory);
+    temp.addEventListener(STEP_INTO_EVENT, onStepInto);
+    temp.addEventListener(STEP_TO_CLUSTER, onStepOut);
+    setController(temp);
```

Comment:
not sure if `deRegisterLayoutFactory` and `deRegisterComponentFactory` are valid or not... if they are then good to use them, right ?
```suggestion
    setController(temp);
    return () => {
     temp.deRegisterLayoutFactory(defaultLayoutFactory);
     temp.deRegisterComponentFactory(stylesComponentFactory);
     temp.removeEventListener(STEP_INTO_EVENT, onStepInto);
     temp.removeEventListener(STEP_TO_CLUSTER, onStepOut);
    }
```

Reply:
removed event listeners but the rest of them are not available.